Sources/Common/DataModel/Collection/index.js

```diff
@@ -0,0 +1,104 @@
+import { max } from 'lodash';
+import macro from 'vtk.js/Sources/macros';
+
+const { vtkErrorMacro } = macro;
+
+// ----------------------------------------------------------------------------
+// vtkCollection methods
+// ----------------------------------------------------------------------------
+
+function vtkCollection(publicAPI, model) {
+  // Set our className
+  model.classHierarchy.push('vtkCollection');
+
+  publicAPI.addItem = (item) => {
+    model.collection.push(item);
+    publicAPI.modified();
+  };
+
+  publicAPI.insertItem = (idx, item) => {
+    model.collection.splice(idx, 0, item);
```

Comment:
Good catch, I will fix that. Additionally, in VTK-C++, the `insertItem` function inserts the item _after_ the provided idx rather than _at_ idx. It therefore also requires that the list be non-empty. I think this behavior is not very intuitive. Do we still want to retain the same behavior in vtk-js?

Reply:
Let's insert *at* the index, and document it as such.